api/v1/helpers/verbs/doPost.js

```diff
@@ -42,9 +42,21 @@ function doPost(req, res, next, props) {
       u.checkDuplicateRLinks(rLinks);
     }
 
-    postPromise = u.getUserNameFromToken(req,
-      featureToggles.isFeatureEnabled('enforceWritePermission'))
-      .then((user) => redisModelSample.postSample(req.swagger.params, user));
+    postPromise = authUtils.getUser(req)
+    .then((user) => {
+      const userObject = user && featureToggles.isFeatureEnabled('returnCreatedBy') ?
+        { name: user.name, id: user.id, email: user.email } : false;
+      return redisModelSample.postSample(req.swagger.params, userObject);
+    })
+    .catch((err) => {
```

Comment:
When there is an error, whether if is with status 403 or not, I am wondering if it makes sense to just reject the request.